// angular2-generic-table/generic-table/components/gt-expanding-row.component.js
"use strict";
var core_1 = require('@angular/core');
var GtExpandedRow = (function () {
    function GtExpandedRow() {
        this.redrawEvent = new core_1.EventEmitter();
    }
    GtExpandedRow.prototype.$hide = function () {
        this.row.isOpen = false;
    };
    GtExpandedRow.prototype.$redraw = function () {
        this.redrawEvent.emit(this.row);
    };
    return GtExpandedRow;
}());
exports.GtExpandedRow = GtExpandedRow;
var GtExpandingRowComponent = (function () {
    function GtExpandingRowComponent() {
        this.redrawEvent = new core_1.EventEmitter();
    }
    GtExpandingRowComponent.prototype.newInstance = function (instance) {
        instance.row = this.row;
        instance.redrawEvent.subscribe(this.redrawEvent);
    };
    GtExpandingRowComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'gt-expanding-row',
                    template: "\n    <div appComponentAnchor\n         [ctor]=\"type\" (instance)=\"newInstance($event)\"></div>"
                },] },
    ];
    /** @nocollapse */
    GtExpandingRowComponent.ctorParameters = function () { return []; };
    GtExpandingRowComponent.propDecorators = {
        'type': [{ type: core_1.Input },],
        'row': [{ type: core_1.Input },],
        'redrawEvent': [{ type: core_1.Output },],
    };
    return GtExpandingRowComponent;
}());
exports.GtExpandingRowComponent = GtExpandingRowComponent;
