// angular2-generic-table/generic-table/components/gt-table-info.component.js
"use strict";
var core_1 = require('@angular/core');
var directives_1 = require('@angular/core/src/metadata/directives');
var GtTableInfoComponent = (function () {
    function GtTableInfoComponent(_changeDetectionRef) {
        this._changeDetectionRef = _changeDetectionRef;
    }
    GtTableInfoComponent.prototype.ngAfterViewChecked = function () {
        this._changeDetectionRef.detectChanges();
    };
    GtTableInfoComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'gt-table-info',
                    template: "<span *ngIf=\"genericTable.gtInfo\">{{(customText? customText:genericTable.gtTexts) | gtTableInfo:genericTable.gtInfo:genericTable.gtInfo.recordsAfterSearch:genericTable.gtInfo.recordFrom:genericTable.gtInfo.recordTo:genericTable.gtInfo.recordsAll:genericTable.gtTexts.loading:genericTable.gtTexts.tableInfoAfterSearch}}</span>"
                },] },
    ];
    /** @nocollapse */
    GtTableInfoComponent.ctorParameters = function () { return [
        { type: core_1.ChangeDetectorRef, },
    ]; };
    GtTableInfoComponent.propDecorators = {
        'genericTable': [{ type: directives_1.Input },],
        'customText': [{ type: directives_1.Input },],
    };
    return GtTableInfoComponent;
}());
exports.GtTableInfoComponent = GtTableInfoComponent;
var core_2 = require('@angular/core');
var TableInfoPipe = (function () {
    function TableInfoPipe() {
    }
    TableInfoPipe.prototype.transform = function (texts, keys, refresh) {
        var text = texts.tableInfo;
        if (keys.recordsAfterSearch !== keys.recordsAll) {
            text = texts.tableInfoAfterSearch;
        }
        for (var key in keys) {
            if (keys.hasOwnProperty(key)) {
                var searchString = new RegExp('#' + key, 'g');
                text = text.replace(searchString, keys[key]);
            }
        }
        return text;
    };
    ;
    TableInfoPipe.decorators = [
        { type: core_2.Pipe, args: [{
                    name: 'gtTableInfo'
                },] },
    ];
    /** @nocollapse */
    TableInfoPipe.ctorParameters = function () { return []; };
    return TableInfoPipe;
}());
exports.TableInfoPipe = TableInfoPipe;
