// angular2-generic-table/generic-table/directives/component-anchor.directive.js
"use strict";
var core_1 = require('@angular/core');
var ComponentAnchorDirective = (function () {
    function ComponentAnchorDirective(componentFactoryResolver, viewContainer) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.viewContainer = viewContainer;
        this.instance = new core_1.EventEmitter();
    }
    ComponentAnchorDirective.prototype.ngOnInit = function () {
        var factory = this.componentFactoryResolver
            .resolveComponentFactory(this.ctor);
        var component = this.viewContainer
            .createComponent(factory, 0, this.injector);
        this.instance.emit(component.instance);
    };
    ComponentAnchorDirective.decorators = [
        { type: core_1.Directive, args: [{
                    selector: '[appComponentAnchor]'
                },] },
    ];
    /** @nocollapse */
    ComponentAnchorDirective.ctorParameters = function () { return [
        { type: core_1.ComponentFactoryResolver, },
        { type: core_1.ViewContainerRef, },
    ]; };
    ComponentAnchorDirective.propDecorators = {
        'ctor': [{ type: core_1.Input },],
        'injector': [{ type: core_1.Input },],
        'instance': [{ type: core_1.Output },],
    };
    return ComponentAnchorDirective;
}());
exports.ComponentAnchorDirective = ComponentAnchorDirective;
